js/csv.js

```diff
@@ -0,0 +1,230 @@
+/**
+ * Reads a CSV file and returns a table with the data.
+ *
+ * @param {Blob|File} file The file object to be read.
+ * @param {Function} parser The function to be used in the
+ * CSV parsing operation.
+ */
+export const readCsv = (file, parser = null) => {
+    parser = parser || parseCsvComplex;
+    return new Promise((resolve, reject) => {
+        const reader = new FileReader();
+        reader.onload = e => resolve(parser(e.target.result));
+        reader.onerror = e => reject(e);
+        reader.readAsText(file);
+    });
+};
+
+/**
+ * Creates the contents of the CSV file with the data given
+ * as an array of rows and the headers.
+ *
+ * @param {Array / Object} data Data to be converted into CSV. Can be
+ * an array of strings, a JSON object, or an array of those.
+ * @param {Array} headers List of headers for the CSV file, is an array of strings.
+ * @param {String} delimiter The delimiter to be used in the CSV.
+ * @returns A string containing the information for the CSV.
+ */
+export const buildCsv = (data, headers = [], delimiter = ",") => {
+    let csv = "";
+    if (headers !== null && Object.keys(headers).length !== 0) csv += headers.toString() + "\n";

```

Comment:
`headers` is always different than null because it has the default value of `[]`

Reply:
It can be explicitly given a `null` value when calling the function, although it does make more sense to simply not invoke those arguments, and write `buildCsv(data)` instead. I'll change it!